### ISSUE TITLE
Detect usage of custom floating-point abs implementation

### ISSUE THREAD

**Issue by krishna-veerareddy** [rust-lang/rust-clippy#5224]
Detect usage of
```rust
if a < 0.0 { // or a.is_sign_negative()
    -a
} else {
    a
}

and

if a > 0.0 { // or a.is_sign_positive()
    a
} else {
    -a
}
```

and suggest usage of `abs` instead
```rust
a.abs()
```

Using `abs` is more efficient, handles more cases and is more succinct than using the if/else block so this lint could go under the `suboptimal_flops` lint. If this seems like an obvious thing and isn't worth the check then I'd be happy to close it.

**Comment by flip1995** [comment]
This should also lint the opposite case:

```rust
if a < 0.0 { // or a.is_sign_negative()
    a
} else {
    -a
}
```

and suggest `-a.abs()`

**Comment by JarredAllen** [comment]
I can work on this. This seems like a good point to begin contributing.

**Comment by flip1995** [comment]
Thanks! If you have questions, just ask here or open a WIP PR. A good read for starting writing lints is this documentation: https://github.com/rust-lang/rust-clippy/blob/master/doc/adding_lints.md
